libpod/image/pull.go

```diff
@@ -56,6 +59,14 @@ type pullStruct struct {
 }
 
 func (ir *Runtime) getPullStruct(srcRef types.ImageReference, destName string) (*pullStruct, error) {
+	imgPart, err := decompose(destName)
+	if err == nil && !imgPart.hasRegistry {
```

Comment:
Surprisingly, `imageParts.hasRegistry` is not actually detecting registry hostnames correctly per the algorithm referred to in 
https://github.com/projectatomic/libpod/issues/667#issuecomment-404939371 ; in `dotless/foo`, `dotless` is a namespace, not a registry name, but `hasRegistry` is set.

That should almost certainly be fixed in `imageParts`, but unless/until that happens, none of the uses of `hasRegistry` in this PR is correct.